SetupDataPkg/Library/ConfigVariableListLib/UnitTest/ConfigVariableListLibUnitTest.c

```diff
@@ -0,0 +1,565 @@
+  )
+{
+  CONFIG_VAR_LIST_ENTRY  *ConfigVarListPtr = NULL;
+  EFI_STATUS             Status;
+  UINT32                 i          = 0;
+  CHAR8                  *AsciiName = NULL;
+
+  for ( ; i < 9; i++) {
+    will_return (GetSectionFromAnyFv, mKnown_Good_Generic_Profile);
+    will_return (GetSectionFromAnyFv, sizeof (mKnown_Good_Generic_Profile));
+    will_return (LibPcdGetPtr, &gZeroGuid);
+
+    ConfigVarListPtr = AllocatePool (sizeof (*ConfigVarListPtr));
+    UT_ASSERT_NOT_NULL (ConfigVarListPtr);
+
+    AsciiName = AllocatePool (StrLen (mKnown_Good_VarList_Names[i]) + 1);
+    UT_ASSERT_NOT_NULL (AsciiName);
+
+    UnicodeStrToAsciiStrS (mKnown_Good_VarList_Names[i], AsciiName, StrLen (mKnown_Good_VarList_Names[i]) + 1);
+
+    Status = QuerySingleActiveConfigAsciiVarList (AsciiName, ConfigVarListPtr);
+    UT_ASSERT_NOT_EFI_ERROR (Status);
+
+    // StrLen * 2 as we compare all bytes, not just number of Unicode chars
+    UT_ASSERT_MEM_EQUAL (mKnown_Good_VarList_Names[i], ConfigVarListPtr->Name, StrLen (mKnown_Good_VarList_Names[i]) * 2);
+    if (i < 2) {
+      UT_ASSERT_MEM_EQUAL (&mKnown_Good_Yaml_Guid, &ConfigVarListPtr->Guid, sizeof (mKnown_Good_Yaml_Guid));
+      UT_ASSERT_EQUAL (3, ConfigVarListPtr->Attributes);
+    } else {
+      // Xml part of blob
+      UT_ASSERT_MEM_EQUAL (&mKnown_Good_Xml_Guid, &ConfigVarListPtr->Guid, sizeof (mKnown_Good_Xml_Guid));
+      UT_ASSERT_EQUAL (7, ConfigVarListPtr->Attributes);
+    }
+
+    UT_ASSERT_EQUAL (mKnown_Good_VarList_DataSizes[i], ConfigVarListPtr->DataSize);
+    UT_ASSERT_MEM_EQUAL (mKnown_Good_VarList_Entries[i], ConfigVarListPtr->Data, ConfigVarListPtr->DataSize);
+
+    FreePool (AsciiName);
+    FreePool (ConfigVarListPtr->Name);
+    FreePool (ConfigVarListPtr->Data);
+    FreePool (ConfigVarListPtr);
+  }
+
+  return UNIT_TEST_PASSED;
+}
+
+/**
+  Unit test for RetrieveActiveConfigVarListTest.
+
+  @param[in]  Context    [Optional] An optional parameter that enables:
+                         1) test-case reuse with varied parameters and
+                         2) test-case re-entry for Target tests that need a
+                         reboot.  This parameter is a VOID* and it is the
+                         responsibility of the test author to ensure that the
+                         contents are well understood by all test cases that may
+                         consume it.
+
+  @retval  UNIT_TEST_PASSED             The Unit test has completed and the test
+                                        case was successful.
+  @retval  UNIT_TEST_ERROR_TEST_FAILED  A test case assertion has failed.
+**/
+UNIT_TEST_STATUS
+EFIAPI
+RetrieveActiveConfigVarListInvalidParamTest (
+  IN UNIT_TEST_CONTEXT  Context
+  )
+{
+  CONFIG_VAR_LIST_ENTRY  *ConfigVarListPtr;
+  EFI_STATUS             Status;
+  UINTN                  ConfigVarListCount;
+
+  Status = RetrieveActiveConfigVarList (NULL, &ConfigVarListCount);
+  UT_ASSERT_STATUS_EQUAL (Status, EFI_INVALID_PARAMETER);

```

Comment:
I think this is fine as is. But i just learnt that if you need to expect an assert in the function call, in the above case, you can do:
`UT_EXPECT_ASSERT_FAILURE (Status = RetrieveActiveConfigVarList (NULL, &ConfigVarListCount), NULL);` where the NULL can be an optional pointer to accept the assert handler.

Reply:
Added, thanks.